modules/debug_cmd/debug_cmd.c

```diff
@@ -160,6 +160,143 @@ static int cmd_play_file(struct re_printf *pf, void *arg)
 }
 
 
+struct ausrc_st;
+
+
+struct fileinfo_st {
+	struct ausrc_st *ausrc;
+	struct ausrc_prm prm;
+	size_t sampc;
+	struct tmr tmr;
+	bool   finished;
+};
+
+
+static void fileinfo_destruct(void *arg)
+{
+	struct fileinfo_st *st = arg;
+
+	tmr_cancel(&st->tmr);
+	mem_deref(st->ausrc);
+}
+
+
+static void fileinfo_timeout(void *arg)
+{
+	struct fileinfo_st *st = arg;
+	size_t ms = 0;
+
+	if (st->prm.ch && st->prm.srate)
+		ms = st->sampc * 1000 / st->prm.ch / st->prm.srate;
+
+	if (st->finished) {
+		info("debug_cmd: length = %u ms\n", ms);
+		ua_event(NULL, UA_EVENT_MAX, NULL, "debug_cmd: length = %u ms",
+				ms);
+	}
+	else if (ms) {
+		warning("debug_cmd: timeout, length > %u ms\n", ms);
+		ua_event(NULL, UA_EVENT_MAX, NULL, "debug_cmd: timeout, "
+				"length > %u ms", ms);
+	}
+	else {
+		info("debug_cmd: timeout\n");
+		ua_event(NULL, UA_EVENT_MAX, NULL, "debug_cmd: timeout");
+	}
+
+	mem_deref(st);
+}
+
+
+static void fileinfo_readh(struct auframe *af, void *arg)
+{
+	struct fileinfo_st *st = arg;
+
+	if (!af || !arg)
+		return;
+
+	st->sampc += af->sampc;
+}
+
+
+static void fileinfo_errh(int err, const char *str, void *arg)
+{
+	struct fileinfo_st *st = arg;
+	(void) err;
+	(void) str;
+
+	st->finished = err ? false : true;
+	tmr_start(&st->tmr, 0, fileinfo_timeout, st);
+}
+
+
+/**
+ * Command fileinfo reads given audio file with ausrc that is specified in
+ * config file_ausrc and returns the length in milli seconds. The file has
+ * to be located in the path specified by audio_path.
+ *
+ * Usage:
+ * /fileinfo audiofile
+ *
+ * @param pf Print handler is used to return length.
+ * @param arg Command argument contains the file name.
+ *
+ * @return The length of the fileplay file in milli seconds.
+ */
+/* ------------------------------------------------------------------------- */
+static int cmd_fileinfo(struct re_printf *pf, void *arg)
+{
+	const struct cmd_arg *carg = arg;
+	int err = 0;
+	size_t len;
+	char *path;
+	char aumod[16];
+	struct fileinfo_st *st = NULL;
+
+	if (!str_isset(carg->prm)) {
+		re_hprintf(pf, "fileplay: filename not specified\n");
+		return EINVAL;
+	}
+
+	err = conf_get_str(conf_cur(), "file_ausrc", aumod, sizeof(aumod));
+	if (err) {
+		warning("debug_cmd: file_ausrc is not set\n");
+		return EINVAL;
+	}
+
+	len = str_len(conf_config()->audio.audio_path) + str_len(carg->prm)
+		+ 2;
+	path = mem_zalloc(len, NULL);
+	re_snprintf(path, len, "%s/%s",
+			conf_config()->audio.audio_path,
+			carg->prm);
```

Comment:
this block can be replaced with `re_sdprintf`

Reply:
cool! ;-)